dask/dataframe/io/io.py

```diff
@@ -499,21 +499,31 @@ def _link(token, result):
     return None
 
 
-def _df_to_bag(df, index=False):
+def _df_to_bag(df, index=False, format="tuple"):
     if isinstance(df, pd.DataFrame):
-        return list(map(tuple, df.itertuples(index)))
+        if format == "tuple":
+            return list(map(tuple, df.itertuples(index)))
+        elif format == "dict":
+            tuple_to_dict = lambda x: dict(x._asdict())
+            return list(map(tuple_to_dict, df.itertuples(index)))
     elif isinstance(df, pd.Series):
-        return list(df.iteritems()) if index else list(df)
+        if format == "tuple":
+            return list(df.iteritems()) if index else list(df)
+        elif format == "dict":
+            return df.to_dict()
```

Comment:
`df.to_dict()` will return a dictionary, instead of a list of dictionaries, when `df` is a pandas `Series`. We'll want to include some additional formatting to make sure we return a list of dictionaries when `format == "dict"`

Reply:
got it, Thanks for pointing out this, I tried the following code for the conversion, let me know if this approach is okay to be proceed 

`
df.to_frame().to_dict(orient="records")
`